packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -174,7 +383,206 @@ class ResidentWebRunner extends ResidentRunner {
+      }
+    }
+    final List<Uri> invalidatedFiles =
+        await ProjectFileInvalidator.findInvalidated(
+      lastCompiled: device.devFS.lastCompiled,
+      urisToMonitor: device.devFS.sources,
+      packagesPath: packagesFilePath,
+    );
+    final Status devFSStatus = logger.startProgress(
+      'Syncing files to device ${device.device.name}...',
+      timeout: timeoutConfiguration.fastOperation,
+    );
+    UpdateFSReport report;
+    try {
+      report = await device.devFS.update(
+        mainPath: mainPath,
+        target: target,
+        bundle: assetBundle,
+        firstBuildTime: firstBuildTime,
+        bundleFirstUpload: isFirstUpload,
+        generator: device.generator,
+        fullRestart: fullRestart,
+        dillOutputPath: dillOutputPath,
+        projectRootPath: projectRootPath,
+        pathToReload: getReloadPath(fullRestart: fullRestart),
+        invalidatedFiles: invalidatedFiles,
+        trackWidgetCreation: true,
+      );
+    } on DevFSException {
+      devFSStatus.cancel();
+      return UpdateFSReport(success: false);
+    }
+    devFSStatus.stop();
+    printTrace('Synced ${getSizeAsMB(report.syncedBytes)}.');
+    return report;
+  }
+
+  @override
+  Future<int> attach({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+  }) async {
+    final Chrome chrome = await ChromeLauncher.connectedInstance;
+    final ChromeTab chromeTab =
+        await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
+      return chromeTab.url.contains(debuggingOptions.hostname);
+    });
+    _wipConnection = await chromeTab.connect();
+    appStartedCompleter?.complete();
+    connectionInfoCompleter?.complete();
+    if (stayResident) {
+      await waitForAppToFinish();
+    } else {
+      await stopEchoingDeviceLog();
+      await exitApp();
+    }
+    await cleanupAtFinish();
+    return 0;
+  }
+}
+
+class _DwdsResidentWebRunner extends ResidentWebRunner {
+  _DwdsResidentWebRunner(
+    FlutterDevice device, {
+    String target,
+    @required FlutterProject flutterProject,
+    @required bool ipv6,
+    @required DebuggingOptions debuggingOptions,
+    bool stayResident = true,
+  }) : super(
+          device,
+          flutterProject: flutterProject,
+          target: target ?? fs.path.join('lib', 'main.dart'),
+          debuggingOptions: debuggingOptions,
+          ipv6: ipv6,
+          stayResident: stayResident,
+        );
+
+  @override
+  Future<int> run({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+    String route,
+  }) async {
+    firstBuildTime = DateTime.now();
+    final ApplicationPackage package = await ApplicationPackageFactory.instance.getPackageForPlatform(
+      TargetPlatform.web_javascript,
+      applicationBinary: null,
+    );
+    if (package == null) {
+      printError('No application found for TargetPlatform.web_javascript.');
```

Comment:
Not sure about the wording. Since we're developing for multiple platforms from a single code base you can get this error even when the application is there. It just may not be configured for `web_javascript`.

Another nit, would developers know what `TargetPlatform.web_javascript` is? It doesn't match https://api.flutter.dev/flutter/foundation/TargetPlatform-class.html that they would be familiar with.

Would something like this be more accurate: "This application is not configured to build for the Web"?

Reply:
That sounds a lot nicer, updated